scrapper/src/test/java/edu/java/client/github/GitHubClientImplTest.java

```diff
@@ -26,18 +30,53 @@ void setUp() {
         gitHubClient = new GitHubClientImpl(baseUrl);
         owner = "owner";
         repo = "repo";
+    }
+
+    @Test
+    void fetchRepo_shouldReturnRepoResponseWithUpdatedAt() {
         String bodyJson = jsonToString("src/test/resources/github.json");
 
         stubFor(get(format("/repos/%s/%s", owner, repo)).willReturn(aResponse()
             .withStatus(200)
             .withHeader("Content-Type", "application/json")
             .withBody(bodyJson)));
+        RepoResponse repoResponse = gitHubClient.fetchRepo(owner, repo);
+
+        assertEquals(OffsetDateTime.parse("2024-02-02T15:17:25Z"), repoResponse.updatedAt());
     }
 
     @Test
-    void fetchRepo_shouldReturnRepoResponseWithUpdatedAt() {
-        RepoResponse repoResponse = gitHubClient.fetchRepo(owner, repo);
+    void fetchCommitsSince() {
+        String bodyJson = jsonToString("src/test/resources/commits-since.json");
 
-        assertEquals(OffsetDateTime.parse("2024-02-02T15:17:25Z"), repoResponse.updatedAt());
+        OffsetDateTime since = OffsetDateTime
+            .of(2024, 1, 22, 6, 5, 8, 0, ZoneOffset.UTC);
+
+        stubFor(get(format("/repos/%s/%s/commits?since=%s", owner, repo, since)).willReturn(aResponse()
+            .withStatus(200)
+            .withHeader("Content-Type", "application/json")
+            .withBody(bodyJson)));
+
+        List<GitHubCommitResponse> gitHubCommitResponses = gitHubClient.fetchCommitsSince(owner, repo, since);
+
+        assertEquals(3, gitHubCommitResponses.size());
```

Comment:
Один коммит можно сделать неподходящим под условие

Reply:
Тут запрос делается с учетом параметра since. Т.е. я замокал что мне должно вернуться если я передам такой since, и если я сделаю один коммит с неподходящим условием мне всеровно будет возвращаться такой запрос

Reply:
А, блин, точняк)